common/v2/services/Store/StoreProvider.tsx

```diff
@@ -59,6 +70,67 @@ export const StoreProvider = ({ children }: { children: React.ReactNode }) => {
     [accounts]
   );
 
+  useEffect(() => {
+    setPendingTransactions(
+      accounts
+        .map(account =>
+          account.transactions.map(tx => {
+            return { ...tx, network: account.network };
+          })
+        )
+        .flatMap(tx => tx)
+        .filter((txObject: ITxReceipt) => txObject.stage === ITxStatus.PENDING) || []
+    );
```

Comment:
1. Here you have a great opportunity to extract a pure function. eg. `setPendingTransactions(pendingTxs)`
By keeping the pure functions separate from side effects it facilitates testing (of the pure function) and facilitates maintainability. 

2. `.map().flatmap(tx => tx)`  is equivalent to `flatmap()`.

For example:
```
  const txIsPending = ({ stage }) => stage === ITxStatus.PENDING
  const pendingTxs = accounts
     .filter(txIsPending) // reduces the size of the list to append network to.
     .flatMap(({transactions: txs, network}) => txs.map(tx => ({...tx, network}))

  
   setPendingTransactions(pendingTxs || [])

```